tree/ntuple/v7/src/RNTupleDescriptor.cxx

```diff
@@ -863,19 +863,15 @@ void ROOT::Experimental::RNTupleDescriptorBuilder::SetNTuple(
    fDescriptor.fGroupUuid = uuid;
 }
 
-void ROOT::Experimental::RNTupleDescriptorBuilder::AddField(
-   DescriptorId_t fieldId, const RNTupleVersion &fieldVersion, const RNTupleVersion &typeVersion,
-   std::string_view fieldName, std::string_view typeName, std::uint64_t nRepetitions, ENTupleStructure structure)
-{
-   RFieldDescriptor f;
-   f.fFieldId = fieldId;
-   f.fFieldVersion = fieldVersion;
-   f.fTypeVersion = typeVersion;
-   f.fFieldName = std::string(fieldName);
-   f.fTypeName = std::string(typeName);
-   f.fNRepetitions = nRepetitions;
-   f.fStructure = structure;
-   fDescriptor.fFieldDescriptors.emplace(fieldId, std::move(f));
+void ROOT::Experimental::RNTupleDescriptorBuilder::AddField(const RFieldDescriptor& fieldDesc) {
+   auto fieldCopy = fieldDesc;
+   fieldCopy.fParentId = kInvalidDescriptorId;
```

Comment:
It might be surprising that we cut the relationship to other field descriptors

Reply:
I definitely agree. I do think we should cut the links (it's very easy to imagine a malformed NTuple otherwise) and leave it up to callers to arrange things with `AddFieldLink`, but am not completely satisfied with the implementation as is.